### PR TITLE
Various bugfixes

### PR DESCRIPTION
**Pull request by Dennisbonke**
This PR fixes various game breaking bugs and implements weapon switching with buttons. It also adds a desktop icon for the game.

### CODE REVIEW

**Review by drfdaniel — APPROVED** [review]
LGTM

File: Assets/Scripts/Core/Utils/WeaponSwitcher.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+		[SerializeField] private Sprite swordButtonTextureSelected_;
+		[SerializeField] private Sprite swordButtonTextureUnselected_;
+
+		[Header("Gun")]
+		[SerializeField] private GameObject gunButton_;
+		[SerializeField] private Sprite gunButtonTextureSelected_;
+		[SerializeField] private Sprite gunButtonTextureUnselected_;
+		public void ChangeActiveWeaponIndicator(int index) {
+			if(index == 0) {
+				swordButton_.GetComponent<Image>().sprite = swordButtonTextureSelected_;
```
Comment (Westlanderz) [inline Assets/Scripts/Core/Utils/WeaponSwitcher.cs:20]: It's not good for the engine to do this, recommend getting the components once on start or awake and storing that in private variables

File: ProjectSettings/ProjectSettings.asset [hunk 2]
```diff
@@ -266,7 +266,49 @@ PlayerSettings:
...
+  m_BuildTargetIcons:
+  - m_BuildTarget: Standalone
+    m_Icons:
+    - serializedVersion: 2
+      m_Icon: {fileID: 2800000, guid: cf2b1b1c158010bfb8fb3e8d0275aa8a, type: 3}
+      m_Width: 1024
+      m_Height: 1024
+      m_Kind: 0
+    - serializedVersion: 2
+      m_Icon: {fileID: 0}
```
Comment (Westlanderz) [inline ProjectSettings/ProjectSettings.asset:278]: These all have fileID 0, I presume that this is correct.

Comment (Dennisbonke) [reply ProjectSettings/ProjectSettings.asset:278]: Unity autogenerated that, so I assume yes. Will find that out soon when luuk starts testing builds.

File: Assets/Scripts/Core/Utils/WeaponSwitcher.cs (hunk 1 above)
Comment (Dennisbonke) [reply Assets/Scripts/Core/Utils/WeaponSwitcher.cs:20]: Noted, I'll PR a fix soon.

**Pull request merged**